docs/disruption_diagram_alternate_structure_examples.md

```diff
@@ -0,0 +1,159 @@
+leans more on the client to determine the appropriate presentation of symbols, line segments, etc.
+rather than directly telling the client what asset to use for each piece of the diagram.
+
+## Example 1: Shuttle from Boston College to Babcock St, screen at Government Center
+![image](assets/disruption_diagrams/1.png)
+
+```ts
+{
+  effect: "shuttle",
+  line: "green",
+  current_station_slot_index: 11,
+  effect_region_slot_index_range: [0, 8],
+  slots: [
+    {type: "terminal", label_id: "place-lake"},
+    {
+      label: {full: "South Street", abbrev: "South St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Chestnut Hill Avenue", abbrev: "Chestnut Hill Ave"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Chiswick Road", abbrev: "Chiswick Rd"},
+      show_symbol: true
+    },
+    {
+      label: "…",
+      show_symbol: false
+    },
+    {
+      label: {full: "Griggs Street", abbrev: "Griggs St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Harvard Avenue", abbrev: "Harvard Ave"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Packard's Corner", abbrev: "Packard's Cn"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Babcock Street", abbrev: "Babcock St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "… via Copley & Kenmore", abbrev: "… via Copley & Kenmore"},
+      show_symbol: false
+    },
+    {
+      label: {full: "Park Street", abbrev: "Park St"},
+      show_symbol: true
+    },
+    {type: "terminal", label_id: "place-gover"}
+  ]
+}
+```
+
+- - -
+
+## Example 2: Suspension from Charles/MGH to South Station, screen at Charles/MGH
+![image](assets/disruption_diagrams/2.png)
+
+(Correction: The first "Downtown Crossing" label should be "Kendall/MIT")
+
+```ts
+{
+  effect: "suspension",
+  line: "red",
+  current_station_slot_index: 2,
+  effect_region_slot_index_range: [3, 4],
+  slots: [
+    {type: "arrow", label_id: "place-alfcl"},
+    {
+      label: {full: "Kendall/MIT", abbrev: "Kendall/MIT"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Charles/MGH", abbrev: "Charles/MGH"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Park Street", abbrev: "Park St"},
+      show_symbol: true
+    },
+    {
+      label: {full: "Downtown Crossing", abbrev: "Downt'n Xng"},
+      show_symbol: true
+    },
+    {
+      label: {full: "South Station", abbrev: "South Sta"},
+      show_symbol: true
+    },
+    {type: "arrow", label_id: "place-asmnl+place-brntn"}
+  ]
+}
+```
+
+- - -
+
+## Example 3: Station closure at Kent St and St. Mary's St, screen at Government Center
+![image](assets/disruption_diagrams/3.png)
+
+```ts
+{
+  effect: "station_closure",
+  line: "green",
+  current_station_slot_index: 11,
+  closed_station_slot_indices: [2, 4],
+  slots: [
+    {type: "arrow", label_id: "place-clmnl"},
+    {
+      label: {full: "Saint Paul Street", abbrev: "St. Paul St"},
```

Comment:
Will these abbreviations be the same that are in `stop.ex`? Or will this widget need its own set of abbreviations?

Reply:
Also, what is the logic for using full vs abbreviations? I see a mix of both in some examples, but no defined logic for which to use when.

Reply:
Yeah, that was my thinking, though I haven't confirmed with Betsy.

> I see a mix of both in some examples, but no defined logic for which to use when.
If you mean the screenshots from my markdown files with example data, I got those from the figma. It does seem to use abbreviation inconsistently, but I assumed that was just because Betsy wasn't focused on getting the labels perfectly right in every example.

According to the figma, the rule is to abbreviate middle-slot labels whenever `label_height > 8 + max(left_end_label_height, right_end_label_height)`

Reply:
++ the abbreviation system is definitely wonky atm. I don't think it will affect the data structure you've chosen, but will add complexity to the serialization / frontend work